builder/xenserver/common/config.go

```diff
@@ -34,6 +34,7 @@ type Config struct {
 	SourcePath        string        `mapstructure:"source_path"`
 
 	Firmware string `mapstructure:"firmware"`
+	SetTemplate string `mapstructure:"set_template"`
```

Comment:
As mentioned in [this comment](https://github.com/ddelnano/packer-plugin-xenserver/issues/89#issuecomment-1672406440), I believe making a template should be the default.

This value is better fit to be a `bool`. That will prevent someone from providing an invalid value (if "true" or "false" is misspelled) and the need to handle that case properly.

In addition to switching the data type, the `mapstructure` library uses Go's implicit default value. This means that for a `bool` member it will default to `false` if it isn't set. Therefore, I think we should rename it so that the default (setting the VM to a template) is applied when it isn't provided.

```suggestion
	SkipSetTemplate string `mapstructure:"skip_set_template"`
```

You can see an example of a `bool` config parameter [here](https://github.com/hashicorp/packer-plugin-amazon/blob/main/builder/common/ami_config.go#L94) in the packer-plugin-amazon.